conjure-java-core/src/integrationInput/java/com/palantir/product/EmptyPathServiceEndpoints.java

```diff
@@ -1,58 +1,74 @@
-public final class EmptyPathServiceEndpoints implements Service {
+public final class EmptyPathServiceEndpoints implements UndertowService {
     private final UndertowEmptyPathService delegate;
 
     private EmptyPathServiceEndpoints(UndertowEmptyPathService delegate) {
         this.delegate = delegate;
     }
 
-    public static Service of(UndertowEmptyPathService delegate) {
+    public static UndertowService of(UndertowEmptyPathService delegate) {
         return new EmptyPathServiceEndpoints(delegate);
     }
 
     @Override
-    public Registrable create(ServiceContext context) {
-        return new EmptyPathServiceRegistrable(context, delegate);
+    public List<Endpoint> endpoints(UndertowRuntime runtime) {
+        return ImmutableList.of(new EmptyPathEndpoint(runtime, delegate));
     }
 
-    private static final class EmptyPathServiceRegistrable implements Registrable {
+    private static final class EmptyPathEndpoint implements HttpHandler, Endpoint {
+        private final UndertowRuntime runtime;
+
         private final UndertowEmptyPathService delegate;
 
-        private final SerializerRegistry serializers;
+        EmptyPathEndpoint(UndertowRuntime runtime, UndertowEmptyPathService delegate) {
+            this.runtime = runtime;
+            this.delegate = delegate;
+        }
 
-        private EmptyPathServiceRegistrable(
-                ServiceContext context, UndertowEmptyPathService delegate) {
-            this.serializers = context.serializerRegistry();
-            this.delegate =
-                    context.serviceInstrumenter()
-                            .instrument(delegate, UndertowEmptyPathService.class);
+        @Override
+        public void handleRequest(HttpServerExchange exchange) throws IOException {
+            boolean result = delegate.emptyPath();
+            runtime.serde().serialize(result, exchange);
         }
 
         @Override
-        public void register(EndpointRegistry endpointRegistry) {
-            endpointRegistry.add(
-                    Endpoint.get("/", "EmptyPathService", "emptyPath"), new EmptyPathHandler());
+        public HttpString method() {
+            return Methods.GET;
         }
 
-        private class EmptyPathHandler implements HttpHandler {
-            @Override
-            public void handleRequest(HttpServerExchange exchange) throws IOException {
-                boolean result = delegate.emptyPath();
-                serializers.serialize(result, exchange);
-            }
+        @Override
+        public String template() {
+            return "/";
+        }
+
+        @Override
+        public Optional<String> serviceName() {
+            return Optional.of("UndertowEmptyPathService");
+        }
+
+        @Override
+        public Optional<String> name() {
+            return Optional.of("emptyPath");
+        }
+
+        @Override
+        public HttpHandler handler() {
+            return this;
```

Comment:
Each endpoint is a fair bit longer than before, however I find this less complex. If we're worried about LoC we can provide an abstract implementation and pass data to the ctor, however then we have to worry about api/abi compatibility on that class as well.